csharp/extractor/Semmle.Extraction.CSharp/Extractor.cs

```diff
@@ -83,22 +83,16 @@ public static ExitCode Run(string[] args)
                 {
                     var compilerVersion = new CompilerVersion(commandLineArguments);
 
-                    bool preserveSymlinks = Environment.GetEnvironmentVariable("SEMMLE_PRESERVE_SYMLINKS") == "true";
-                    var canonicalPathCache = CanonicalPathCache.Create(logger, 1000, preserveSymlinks ? CanonicalPathCache.Symlinks.Preserve : CanonicalPathCache.Symlinks.Follow);
-
-                    if (compilerVersion.SkipExtraction)
-                    {
-                        logger.Log(Severity.Warning, "  Unrecognized compiler '{0}' because {1}", compilerVersion.SpecifiedCompiler, compilerVersion.SkipReason);
-                        return ExitCode.Ok;
-                    }
-
                     var cwd = Directory.GetCurrentDirectory();
+
+#nullable disable
```

Comment:
We can't do a lot about this. `CSharpCOmmandLineParser.Parse` defines `string additionalReferenceDirectories = null` as the last parameter.